llvm/lib/Target/AMDGPU/GCNSubtarget.h

```diff
@@ -1254,12 +1254,6 @@ class GCNSubtarget final : public AMDGPUGenSubtargetInfo,
   // \returns true if preloading kernel arguments is supported.
   bool hasKernargPreload() const { return KernargPreload; }
 
-  // \returns true if we need to generate backwards compatible code when
-  // preloading kernel arguments.
-  bool needsKernargPreloadBackwardsCompatibility() const {
```

Comment:
Isn't this still needed if you're just adding a new debug trap?

Reply:
It will be needed but is unused at the moment; can be added back.